From ab4d3d6fab1b09e536f0c25234437428deb878f4 Mon Sep 17 00:00:00 2001
From: Yosef Or Boczko <yosefor3@walla.com>
Date: Tue, 11 Jun 2013 00:42:11 +0300
Subject: Use GtkButton subclasses instead of GdHeaderButton ones

and add close button to headerbar and fix arrow icons in RTL locales
---
 src/dh-window.c | 64 +++++++++++++++++++++++++++++++++++++++------------------
 1 file changed, 44 insertions(+), 20 deletions(-)

diff --git a/src/dh-window.c b/src/dh-window.c
index abec66d..8029e88 100644
--- a/src/dh-window.c
+++ b/src/dh-window.c
@@ -650,46 +650,70 @@ window_populate (DhWindow *window)
 {
         DhWindowPriv  *priv;
         DhBookManager *book_manager;
-        GtkWidget     *back;
-        GtkWidget     *forward;
+        GtkWidget     *back_button;
+        GtkWidget     *back_image;
+        GtkWidget     *forward_button;
+        GtkWidget     *forward_image;
         GtkWidget     *box;
         GtkWidget     *menu_button;
+        GtkWidget     *menu_image;
+        GtkWidget     *separator;
+        GtkWidget     *close_button;
+        GtkWidget     *close_image;
         GObject       *menu;
+        const char *prev_icon, *next_icon;
 
         priv = window->priv;
         book_manager = dh_app_peek_book_manager (DH_APP (gtk_window_get_application (GTK_WINDOW (window))));
 
         priv->header_bar = gtk_header_bar_new ();
 
-        back = gd_header_simple_button_new ();
-        gd_header_button_set_label (GD_HEADER_BUTTON (back),
-                                    _("Back"));
-        gd_header_button_set_symbolic_icon_name (GD_HEADER_BUTTON (back),
-                                                 "go-previous-symbolic");
-        gtk_actionable_set_action_name (GTK_ACTIONABLE (back), "win.go-back");
-
-        forward = gd_header_simple_button_new ();
-        gd_header_button_set_label (GD_HEADER_BUTTON (forward),
-                                    _("Forward"));
-        gd_header_button_set_symbolic_icon_name (GD_HEADER_BUTTON (forward),
-                                                 "go-next-symbolic");
-        gtk_actionable_set_action_name (GTK_ACTIONABLE (forward), "win.go-forward");
+        if (gtk_widget_get_direction (GTK_WIDGET (window)) == GTK_TEXT_DIR_RTL) {
+                prev_icon = "go-previous-rtl-symbolic";
+                next_icon = "go-next-rtl-symbolic";
+        } else {
+                prev_icon = "go-previous-symbolic";
+                next_icon = "go-next-symbolic";
+        }
+        
+        back_button = gtk_button_new();
+        back_image = gtk_image_new_from_icon_name(prev_icon, GTK_ICON_SIZE_MENU);
+        gtk_button_set_image(GTK_BUTTON(back_button), back_image);
+        gtk_widget_set_tooltip_text(back_button, _("Back"));
+        gtk_actionable_set_action_name (GTK_ACTIONABLE (back_button), "win.go-back");
+
+        forward_button = gtk_button_new();
+        forward_image = gtk_image_new_from_icon_name(next_icon, GTK_ICON_SIZE_MENU);
+        gtk_button_set_image(GTK_BUTTON(forward_button), forward_image);
+        gtk_widget_set_tooltip_text(forward_button, _("Forward"));
+        gtk_actionable_set_action_name (GTK_ACTIONABLE (forward_button), "win.go-forward");
 
         box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
         gtk_style_context_add_class (gtk_widget_get_style_context (box), "linked");
-        gtk_box_pack_start (GTK_BOX (box), back, FALSE, FALSE, 0);
-        gtk_box_pack_start (GTK_BOX (box), forward, FALSE, FALSE, 0);
+        gtk_box_pack_start (GTK_BOX (box), back_button, FALSE, FALSE, 0);
+        gtk_box_pack_start (GTK_BOX (box), forward_button, FALSE, FALSE, 0);
         gtk_header_bar_pack_start (GTK_HEADER_BAR (priv->header_bar), box);
 
-        menu_button = gd_header_menu_button_new ();
-        gd_header_button_set_symbolic_icon_name (GD_HEADER_BUTTON (menu_button),
-                                                 "emblem-system-symbolic");
+        menu_button = gtk_menu_button_new();
+        menu_image = gtk_image_new_from_icon_name("emblem-system-symbolic", GTK_ICON_SIZE_MENU);
+        gtk_button_set_image(GTK_BUTTON(menu_button), menu_image);
         gtk_actionable_set_action_name (GTK_ACTIONABLE (menu_button), "win.gear-menu");
 
         gtk_header_bar_pack_end (GTK_HEADER_BAR (priv->header_bar), menu_button);
 
         menu = gtk_builder_get_object (priv->builder, "window-menu");
         gtk_menu_button_set_menu_model (GTK_MENU_BUTTON (menu_button), G_MENU_MODEL (menu));
+        
+        separator = gtk_separator_new(GTK_ORIENTATION_VERTICAL);
+        gtk_header_bar_pack_end(GTK_HEADER_BAR (priv->header_bar), separator);
+        
+        close_button = gtk_button_new();
+        gtk_button_set_relief(GTK_BUTTON(close_button), GTK_RELIEF_NONE);
+        close_image = gtk_image_new_from_icon_name("window-close-symbolic", GTK_ICON_SIZE_MENU);
+        gtk_button_set_image(GTK_BUTTON(close_button), close_image);
+        gtk_actionable_set_action_name (GTK_ACTIONABLE (close_button), "win.close");
+        
+        gtk_header_bar_pack_end(GTK_HEADER_BAR (priv->header_bar), close_button);
 
         /* Make headerbar a titlebar */
         gtk_window_set_titlebar(GTK_WINDOW(window), priv->header_bar);
-- 
1.8.3

